src/map/script.cpp

```diff
@@ -26528,6 +26528,34 @@ BUILDIN_FUNC(item_enchant){
 #endif
 }
 
+/**
+* Generate item link string for client
+* itemlink(<item_id>,<refine>,<card0>,<card1>,<card2>,<card3>,<enchantgrade>{,<RandomIDArray>,<RandomValueArray>,<RandomParamArray>});
+* @author [Cydh]
+**/
+BUILDIN_FUNC(itemlink)
+{
+	struct item item = {};
+	item.nameid = script_getnum(st, 2);
+
+	FETCH(3, item.refine);
+	FETCH(4, item.card[0]);
+	FETCH(5, item.card[1]);
+	FETCH(6, item.card[2]);
+	FETCH(7, item.card[3]);
+	FETCH(8, item.enchantgrade);
+
+#if PACKETVER >= 20150225
+	if ( script_hasdata(st,9) && script_getitem_randomoption(st, nullptr, &item, "itemlink", 9) == SCRIPT_CMD_FAILURE ) {
```

Comment:
Return empty string (or end the script)

Reply:
im not sure what you mean, sorry. Could you point me in the right direction?

Reply:
The npc script is still running when SCRIPT_CMD_FAILURE is returned. But the itemlink function is supposed to return a string.
So return an empty string via `script_pushconststr(st,"");`
or stop the script with `st->state = END;` (before SCRIPT_CMD_FAILURE)

Reply:
done 😊